### PR TITLE
[ci-app] Handle Uncaught Async Errors in Mocha 

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
* Fix bug where uncaught async errors are not instrumented correctly. 

### Motivation
Be able to instrument errors that are uncaught, such as: 

```javascript
it('will fail asynchronously', (done) => {
  setTimeout(() => {
    expect(true).to.equal(false)
    done()
  }, 200)
})
```

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/datadog-plugin-mocha/src/index.js [hunk 1]
```diff
@@ -211,6 +238,7 @@ module.exports = [
     unpatch (Runner) {
       this.unwrap(Runner.prototype, 'runTests')
       this.unwrap(Runner.prototype, 'runTest')
+      this.unwrap(Runner.prototype, 'fail')
```
Comment (juan-fernandez) [inline packages/datadog-plugin-mocha/src/index.js:241]: unrelated fix

**Review by rochdev — APPROVED** [review]
